modules/apigee_edge_apidocs/tests/src/Kernel/ApiDocTest.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\Tests\content_entity_example\Kernel;
+
+use Drupal\apigee_edge_apidocs\Entity\ApiDoc;
+use Drupal\KernelTests\KernelTestBase;
+
+/**
+ * Test basic CRUD operations for our ApiDoc entity type.
+ *
+ * @group apigee_edge
+ * @group apigee_edge_apidocs
+ */
+class ApiDocTest extends KernelTestBase {
+
+  protected static $modules = ['user', 'text', 'file', 'apigee_edge_apidocs'];
+
+  /**
+   * Basic CRUD operations on a ApiDoc entity.
+   */
+  public function testEntity() {
+    $this->installEntitySchema('apidoc');
+    $entity = ApiDoc::create([
+      'name' => 'API 1',
+      'description' => 'Test API 1',
+      'spec' => NULL,
```

Comment:
An API Doc may not need a spec when setting up a doc that starts only with a summary when first configuring.

Reply:
This seems to something that should be solved by using Workflow module. The Workflow states would specify whether a field is required or not. So you could create new API documentation with a "draft" state without attaching an API documentation file but you can not change its state to "published" without attaching the actual the actual API specification to the entity. 
We have built plenty of portals with an implementation like this to customers.

https://www.drupal.org/docs/8/core/modules/workflows/overview

Reply:
It seems like this isn't really something we can solve right now. If a field is required, you cannot save an entity, but as described by @cnovak there is a valid usecase for it. Given that I mark this as resolved. I think it could be up to the individual site owners to implement custom workflows if they need some.